### PR TITLE
macOS: Support set_activation_policy during runtime

### PR DESCRIPTION
**Pull request by probablykasper**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information

Example usage:

```rust
event_loop.run(move |event, event_loop_wt, control_flow| {
  // ...
  #[cfg(target_os = "macos")]
  event_loop_wt.set_activation_policy(ActivationPolicy::Accessory)
}
```

### CODE REVIEW

**Comment by amrbashir** [comment]
you mentioned that the window will be hidden when switching the activation policy to Accessory, is this still the case ?

**Comment by probablykasper** [comment]
Yes, as far as I can tell that's just how macOS handles it

**Comment by amrbashir** [comment]
Not according to the docs.
> from: https://developer.apple.com/documentation/appkit/nsapplicationactivationpolicy/nsapplicationactivationpolicyaccessory?language=objc
> The application doesn’t appear in the Dock and doesn’t have a menu bar, but it may be activated programmatically or by clicking on one of its windows.

**Comment by probablykasper** [comment]
I can't see anything about the behavior at all in Apple's documentation, so it just doesn't seem to be documented.

[This SO answer](https://stackoverflow.com/a/68057340/6553404) mentions the behavior:
>  2. App windows **went backward/hidden** after App icon in Dock is **disabled** (after setting `NSApplication.ActivationPolicy.accessory`). You need to launch "Mission control" to reveal App windows.

**Comment by amrbashir** [comment]
tbh I don't think this brings much value and the main reason was to hide/show dock icon, so I think we should just close this and look for a proper way to do it by following electron implementation.

**Comment by probablykasper** [comment]
Was the main reason for adding set_activation_policy to show/hide the Dock icon?

I do think this would be very useful for an overlay search bar that wants to toggle a settings window

**Comment by amrbashir** [comment]
> Was the main reason for adding set_activation_policy to show/hide the Dock icon?

yeah and we already have `set_activation_policy()` it just doesn't work at runtime, yours does but not perfect, I guess? that's why I feel it is not worth the effort and people would be confused as to why we have two methods with the same name. 
 
> I do think this would be very useful for an overlay search bar that wants to toggle a settings window

sure, but we should implement dock management the right way.

**Comment by probablykasper** [comment]
> confused as to why we have two methods

Should the original one be removed? Or is there some way to make it only one method?

> sure, but we should implement dock management the right way.

I'm not really talking about toggling the dock icon, I mean that's a use case for switching the activation policy on runtime. The activation policy API isn't there just for toggling the dock icon

**Comment by amrbashir** [comment]
> > confused as to why we have two methods
> 
> Should the original one be removed? Or is there some way to make it only one method?

No, lets keep it and change this one to a different name, `set_activation_policy_at_runtime()` and you should probably document any bugs it has if any, when called during runtime.

> > sure, but we should implement dock management the right way.
> 
> I'm not really talking about toggling the dock icon, I mean that's a use case for switching the activation policy on runtime. The activation policy API isn't there just for toggling the dock icon

Ah okay.

**Comment by probablykasper** [comment]
Renamed it to that, though maybe it would make more sense for the runtime one to be `set_activation_policy`, and the pre-runtime one (which isn't calling `setActivationPolicy`) to be `with_activation_policy`.

No bugs as far as I can tell.

**Comment by slantview** [comment]
👍  Need this asap, thanks for the work!

File: examples/system_tray.rs [hunk 1]
```diff
@@ -30,6 +33,13 @@ fn main() {
   // open new window menu item
   let open_new_window_element = submenu.add_item(MenuItemAttributes::new("Open new window"));
 
+  #[cfg(target_os = "macos")]
+  let mut is_accessory = false;
+  // Accessory activation policy toggle
+  #[cfg(target_os = "macos")]
+  let mut activation_policy_accessory =
+    submenu.add_item(MenuItemAttributes::new("Accessory Activation Policy").with_selected(false));
```
Comment (amrbashir) [inline examples/system_tray.rs]: No need to add an example of it since it is platform-specific.

File: src/platform/macos.rs [hunk 2]
```diff
@@ -390,11 +390,11 @@ impl WindowBuilderExtMacOS for WindowBuilder {
 }
 
 pub trait EventLoopExtMacOS {
-  /// Sets the activation policy for the application. It is set to
+  /// Sets the initial activation policy for the application. It is set to
```
Comment (amrbashir) [inline src/platform/macos.rs]:
```suggestion
  /// Sets the activation policy for the application. It is set to
```
Lets keep this as is.

**Review by amrbashir — APPROVED** [review]

File: examples/system_tray.rs (hunk 1 above)
Comment (kasper9n) [reply examples/system_tray.rs]: Removed it

**Comment by amrbashir** [comment]
Next time please wait for @wusyong to approve and merge macOS specific PRs.

My approval of macOS PRs should not be considered as a green-light for merge.

**Comment by lucasfernog** [comment]
Oops, changed the branch protection rules for `next`. Thanks @amrbashir

**Comment by probablykasper** [comment]
Ah sorry, will keep that in mind

**Pull request merged**